src/model/implementation/field.ts

```diff
@@ -1067,11 +1067,15 @@ export class Field implements ModelComponent {
                 )
             );
         }
-        if (this.isIncludedInSearch && !(this.type.isScalarType && this.type.name === 'String')) {
+        if (
+            this.isIncludedInSearch &&
+            !this.type.isObjectType &&
```

Comment:
ah I think root entity types (relations and references) should also not be allowed, neither should `@collect` fields (they don't actually exist in the database)

Reply:
relations, references and collect fields cannot be flexSearchIndexed anyway so they cannot be "includedInSearch" anyway.

Reply:
aah right this is only for `includedInSearch`.